lib/krane/cluster_resource_discovery.rb

```diff
@@ -37,12 +37,43 @@ def fetch_resources(namespaced: false)
       end.compact.uniq { |r| "#{r['apigroup']}/#{r['kind']}" }
     end
 
+    def fetch_group_kinds
+      output, err, st = kubectl.run("api-resources", "--no-headers=true", attempts: 2, use_namespace: false)
+      if st.success?
+        output.split("\n").map do |l|
+          matches = l.scan(/\S+/)
+
+          if matches.length == 4
+            group = ::Krane::KubernetesResource.group_from_api_version(matches[1])
+            # name, api, namespaced, kind
+            {
+              "namespaced" => matches[2] == "true",
+              "group" => group,
+              "kind" => matches[3],
+              "group_kind" => ::Krane::KubernetesResource.combine_group_kind(group, matches[3]),
+            }
+          else
+            group = ::Krane::KubernetesResource.group_from_api_version(matches[2])
+            # name, shortname, api, namespaced, kind
+            {
+              "namespaced" => matches[3] == "true",
+              "group" => group,
+              "kind" => matches[4],
+              "group_kind" => ::Krane::KubernetesResource.combine_group_kind(group, matches[4]),
```

Comment:
Is it worth making a proper class (`APIResource`?) out of this, with something like a `::Parse` method. That way we can encapsulate the initialization as well as provide a way to unit test it.